src/chapters/style/documentation.typ

```diff
@@ -0,0 +1,247 @@
+- multiple `term` items for parameter type hints and descriptions,
+- a return type hint `-> type`,
+- and multiple property annotations (`#property("private")` or `#property("deprecated")`).
+
+Types in type lists or return type annotations may be separated by `|` to indicate that more than one type is accepted, the exact types allowed depend on the doc parser, but the built in types are generally supported.
+
+Parameter description and return types (if present) are placed tightly together, property annotations if present are separated using another empty doc comment line.
+
+Parameter documentation is created by writing a term item containing the parameter name, a mandatory type list in parentheses and optional description.
+If the parameter is an argument sink its name must likewise contain the spread operator `..`.
+Each parameter can only be documented once, but doesn't have to, undocumented parameters are considered private by convention and may only optional parameters.
+
+#do-dont[
+  ```typst
+  /// Function description
+  ///
+  /// / b (any): b descpription
+  /// / a (int | float): a description
+  #let func(a, b, c) = { ... }
+
+  /// Function description
+  ///
+  /// / ..args (int): args description
+  #let func(..args) = { ... }
+  ```
+][
+  ```typst
+  /// Function description
+  ///
+  /// / c (any): c doesn't exist
+  /// / a (int | float): a description
+  #let func(a, b) = { ... }
+
+  /// Missing empty line between function description and trailer
+  /// / b (any): b descpription
+  /// / a (int | float): a description
+  #let func(a, b) = { ... }
+
+
+  /// Function description
+  ///
+  /// / b (any): b descpription
+  /// / a: missing types
+  #let func(a, b) = { ... }
+
+  /// Function description
+  ///
+  /// / args (int): missing spread operator for args
+  #let func(..args) = { ... }
+  ```
+]
+
+The return type can only be annotated once, on a single line after all parameters, if any exist.
+For non function types the return type annotation can be used as a normal type annotation.
+Items which bind a `function.with` expression should be treated as regular function definitions, i.e. their return type is the return type of the function when called, not `function` itself as would be the return value of `function.with`.
+
+#do-dont[
+  ```typst
+  /// Function doc
+  ///
+  /// / arg (types): Description for arg
+  /// -> types
+  #let func(arg) = { ... }
+
+  /// Function doc
+  ///
+  /// -> types
+  #let other = func.with(default)
+
+  /// Value doc
+  ///
+  /// -> type
+  #let value = { ... }
+  ```
+][
+  ```typst
+  /// Missing empty line between description trailer
+  ///
+  /// -> type
+  /// / arg (types): arg descrption after return type
+  #let func(arg) = { ... }
+  ```
+]
+
+Property annotations can be used to document package specific or otherwise important information like deprecation status, visibility or contextuality.
```

Comment:
> or otherwise important information

what does _otherwise_ mean?

Reply:
Something that is _not_ package-specific, i.e. deprecation status and visibility, can be considered standard annotations. Whereas something like cetz could define its own annotations for cetz-specific things like how a function interacts with its internal style system.

I assume it's the phrasing that's problematic here so I'll rephrase it.